change_method_utils.py

```diff
@@ -0,0 +1,111 @@
+import os
+import requests
+
+from bs4 import BeautifulSoup
+
+import utils
+
+log_dir = os.environ.get('PY_GOOGLE_AUTH_LOG_PATH')
+
+
+def get_payload_for_select_page(form_html, form_key):
+    '''
+    Function to collect payload from method selection page.
+    The page contains multiple forms with hidden fields having payload params.
+    using `form_key` we select appropriate form and then collect input field values.
+    You can read payload field details in `get_payload` function.
+    '''
+
+    payload = {}
+
+    # find all the forms
+    forms = BeautifulSoup(form_html).find_all('form')
+
+    # collect all inputs from the appropriate form, these contains the parameters for payload.
+    inputs = forms[form_key].find_all('input')
+
+    # create a payload dictionary
+    for u in inputs:
+        if u.has_attr('value') and u.has_attr('name'):
+            payload[u['name']] = u['value']
+
+    return payload
+
+
+def get_method_for_selection(selected_method):
+    '''
+    Returns two word method name for a complete method string given by google while selecting
+    alternate tfa method.
+    This is done because front end is using these two word method names to render forms for each
+    method.
+    '''
+    methods = utils.get_method_names()
+    method = [method for method in methods if methods[method][0] in selected_method][0]
+    return method
+
+
+def get_alternate_method(session, method, select_challenge_url):
+    '''
+    Function to get the url for alternatively selected method from the form in try another
+    method page.
+    '''
+
+    error = None
```

Comment:
because as you can see, none of the try block has `error` defined. So if no `except` blocks were to be executed then it will cause `error`: name not defined, since the last line of the method refers `error` in either of the cases.